### ISSUE TITLE
elgg_get_plugin_settings() doesn't respect default value if any settings are cached

### ISSUE THREAD

**Issue by brettp** [Elgg/Elgg#9781]
```php
	public function getSetting($name, $default = null) {
		$values = _elgg_services()->pluginSettingsCache->getAll($this->guid);

		if ($values !== null) {
			return isset($values[$name]) ? $values[$name] : null;
		}
```

If any settings have been saved or are cached, the if block ignores the default values, even if the cached settings don't include the setting we're requesting.

**Comment by mrclay** [comment]
Oops my bad.

**Comment by mrclay** [comment]
I rebased your fix on 2.1: #9788
